csrc/device_lower/analysis/fusion_info.h

```diff
@@ -0,0 +1,118 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#pragma once
+
+#include <compute_at_map.h>
+#include <device_lower/analysis/fused_reduction.h>
+#include <device_lower/analysis/padded_parallel_dimensions.h>
+#include <device_lower/analysis/thread_predicate.h>
+#include <device_lower/analysis/trivial_broadcast.h>
+#include <id_model/id_model.h>
+#include <parallel_dimension_map.h>
+
+#include <memory>
+
+namespace nvfuser {
+
```

Comment:
I'd appreciate adding a comment summarizing the purpose and plan for `FusionInfo` here.

1. Are there any other known constraints for using `FusionInfo`?  e.g., You need to have a scheduled fusion to complete the analysis.
2. Which passes are supported and will be in the future?
3. A note about handling the prerequisite passes for each analysis pass.